src/storage/data_table.cpp

```diff
@@ -881,19 +1053,49 @@ void DataTable::VerifyUpdateConstraints(TableCatalogEntry &table, DataChunk &chu
 			break;
 		}
 		case ConstraintType::UNIQUE:
-		case ConstraintType::FOREIGN_KEY:
 			break;
+		case ConstraintType::FOREIGN_KEY: {
+			auto &bfk = *reinterpret_cast<BoundForeignKeyConstraint *>(constraint.get());
+			if (bfk.info.type == ForeignKeyType::FK_TYPE_SELF_REFERENCE_TABLE) {
+#ifdef DEBUG
+				is_self_referencing_fk = true;
+#endif
+				bool is_verify_append = false, is_verify_delete = false;
+				for (idx_t i = 0; i < column_ids.size(); i++) {
+					if (bfk.fk_key_set.find(column_ids[i]) != bfk.fk_key_set.end()) {
+						is_verify_append = true;
+					}
+					if (bfk.pk_key_set.find(column_ids[i]) != bfk.pk_key_set.end()) {
+						is_verify_delete = true;
+					}
+				}
+				if (is_verify_append) {
+					DataChunk mock_chunk;
+					auto types = table.GetTypes();
+					CreateMockChunk(types, column_ids, chunk, mock_chunk);
+					VerifyAppendForeignKeyConstraint(bfk, context, mock_chunk);
+				}
+				if (is_verify_delete) {
+					DataChunk verify_chunk;
+					GatherVerifyChunk(context, row_ids, count, verify_chunk);
```

Comment:
This chunk, and `GatherVerifyChunk`, appear to never be called. Could you add a test case that tests them?